numpy/lib/_function_base_impl.py

```diff
@@ -5732,6 +5732,13 @@ def append(arr, values, axis=None):
     insert : Insert elements into an array.
     delete : Delete elements from an array.
 
+    Notes
+    -----
+    If dtype of `values` is different from the dtype of `arr` then
+    their dtypes are compared to figure out the common dtype they can
+    both be safely coerced to. For e.g. `int64` and `float64` can both
+    go to `float64`. See examples section below.
+
```

Comment:
I think either the example or the note is sufficient, no need for both, and I prefer the example.